pkg/inspector/image-inspector.go

```diff
@@ -298,3 +324,36 @@ func getAuthConfigs(dockercfg, username, password_file string) (*docker.AuthConf
 
 	return imagePullAuths, nil
 }
+
+func (i *defaultImageInspector) scanImage(s openscap.Scanner) ([]byte, error) {
+	log.Printf("%s scanning %s. Placing results in %s",
+		s.ScannerName(), i.opts.DstPath, i.opts.ScanResultsDir)
+	err := s.Scan(i.opts.DstPath, &i.meta.Image)
+	if err != nil {
+		return []byte(""), fmt.Errorf("Unable to run %s: %v\n", s.ScannerName(), err)
+	}
+	scanReport, err := ioutil.ReadFile(s.ResultsFileName())
+	if err != nil {
+		return []byte(""), fmt.Errorf("Unable to read %s result file: %v\n", s.ScannerName(), err)
+	}
+	return scanReport, nil
+}
+
+func createOutputDir(dirName string, tempName string) (string, error) {
+	if len(dirName) > 0 {
+		_, err := os.Stat(dirName)
+		if os.IsNotExist(err) {
```

Comment:
@enoodle I really don't understand why you had to change this code, without a proper comment in the commit, without mentioning it anywhere, and even introducing a bug with a race :smile: 

There was a reason for the previous code.